Unpinning Cryptography

freyes opened the pull request:
Fixes #421

freyes committed:
Unpin cryptography

freyes committed:
Unpin hvac

The commit 29944df pinned hvac to `<0.7.0`, because it was breaking the
tests, this is no longer necessary.

freyes committed:
Build all packages from source

Pass `--no-binary ":all:"` to the tox install_command, so all
dependencies get built from sources, this allows to detect possible
issues when using zaza in non-x86 environments where there is no
published binary wheel in pypi.

freyes committed:
Drop python 3.11 from the testing matrix

freyes committed:
Pin setuptools_scm < 8.2.0

freyes committed:
Pin importlib-metadata<8.0.0

This is the last version compatible with Python 3.8

https://pypi.org/project/importlib-metadata/7.2.1/

chanchiwai-ray commented:
Hi @freyes, thanks for working on this, and we would like to see this eventually unpinned.  For the issue seen in the CI, we've seen that before for python 3.8, and it worked well if "setuptool_scm<8.2.0" is in the **constraints** (instead of a requirement) see https://github.com/canonical/charm-logrotated/pull/72 for example. Hope that helps

freyes committed:
Pin pydocstyle earlier.

Pinning pydocstyle before flake8 allows us to control the version,
otherwise flake8 pulls in a newer version.
